provider/mesh/peer.go

```diff
@@ -0,0 +1,273 @@
+package mesh
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/prometheus/alertmanager/provider"
+	"github.com/prometheus/alertmanager/types"
+	"github.com/prometheus/common/log"
+	"github.com/prometheus/common/model"
+	"github.com/satori/go.uuid"
+	"github.com/weaveworks/mesh"
+)
+
+// NotificationInfos provides gossiped information about which
+// receivers have been notified about which alerts.
+type NotificationInfos struct {
+	st     *notificationState
+	send   mesh.Gossip
+	logger log.Logger
+}
+
+func NewNotificationInfos(logger log.Logger) *NotificationInfos {
```

Comment:
On 22 June 2016 at 13:15, Fabian Reinartz <notifications@github.com> wrote:

> It says "should non-trivial unexported type or function declarations" –
> "should" and "non-trivial". Triviality was exactly my argument.


But you forgot the "unexported". Non-triviality only applies to unexported
type or function declarations. The function my review comment was about is
exported. golint never complains about doc comments for unexported type or
function declaration, exactly because it is not able to judge about
triviality.


-- 
Björn Rabenstein, Engineer
http://soundcloud.com/brabenstein

SoundCloud Ltd. | Rheinsberger Str. 76/77, 10115 Berlin, Germany
Managing Director: Alexander Ljung | Incorporated in England & Wales with
Company No. 6343600 | Local Branch Office | AG Charlottenburg  | HRB 110657B